### PR TITLE
Solving Issue 37, changing scopes of major keywords, changing scope o…

### PR DESCRIPTION
**Pull request by anubmat**
Solving Issue #37
- Changed scopes of major keywords
- Changed scope of switch statement
- Modified type-annotation to allow 'typeof' keyword

### CODE REVIEW

File: TypeScript.YAML-tmLanguage [hunk 1]
```diff
@@ -490,8 +492,12 @@ repository:
     - include: '#expression'
 
   expression-operator:
-    name: keyword.operator.ts
-    match: =>|\b(delete|export|import|from|in|instanceof|new|typeof|void|as)\b
+    name: keyword.others.ts
+    match: =>|\b(delete|in|instanceof|new|typeof|as|is|of)\b
```
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: Yield should go here

File: TypeScript.YAML-tmLanguage [hunk 2]
```diff
@@ -511,7 +517,7 @@ repository:
 
   storage-keyword:
     name: storage.type.ts
-    match: \b(var|let|function|const|module|namespace)\b
+    match: \b(var|let|function|const|module|namespace|void|export)\b
```
Comment (DanielRosenwasser) [inline TypeScript.YAML-tmLanguage]: There are also `void` expressions so I would move those out.

File: TypeScript.YAML-tmLanguage (hunk 1 above)
Comment (Gheoan) [inline TypeScript.YAML-tmLanguage]: Shouldn't `yield` be `control-statement` like in #117 ?

File: tests/cases/Issue37.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+var m: ^^typeof M = M;
+
+var adder = ^^new Adder(3, 4);
+^^delete (adder)
+
+var s = typeof {}
+interface I {}
+var j: I = {}
```
Comment (zhengbli) [inline tests/cases/Issue37.ts]: Can you add the original issue repro text in #37 to this test? namely
```ts
module Body {
    export class Cell {
        public parent: Cell;

        constructor(parent: Cell) {
            this.parent = parent;
            if (typeof parent === 'object' || parent instanceof Error) {
            }
        }
```

File: TypeScript.YAML-tmLanguage (hunk 1 above)
Comment (anubmat) [inline TypeScript.YAML-tmLanguage]: Lets consider the keyword 'yield' as a separate pull request for issue #29.

**Comment by aeschli** [comment]
I tried the changes and they look good to me, with two exceptions: `=>` and `=` for default parameters are now also keyword.other. From a coloring perspective I'd suggest to keep them as keyword.operand.

File: TypeScript.YAML-tmLanguage [hunk 4]
```diff
@@ -339,7 +339,7 @@ repository:
     name: meta.type.function.return.ts
     begin: '=>'
     beginCaptures:
-      '0': { name: keyword.others.ts }
+      '0': { name: keyword.operator.ts }
```
Comment (zhengbli) [inline TypeScript.YAML-tmLanguage]: @aeschli suggested to make to [`keyword.operand`](https://github.com/Microsoft/TypeScript-TmLanguage/pull/105#issuecomment-195747412), does it make a difference? Or maybe he meant `operator`

Comment (anubmat) [inline TypeScript.YAML-tmLanguage]: There was no field as keyword.operand. And then he said -- keep them as --. 
Considering the definitions of operator and operand I felt he probably meant operator, which they were earlier.

**Comment by zhengbli** [comment]
:+1: 
Also tested using Sublime, the coloring looks good to me.

File: TypeScript.YAML-tmLanguage (hunk 4 above)
Comment (aeschli) [inline TypeScript.YAML-tmLanguage]: Sorry, `keyword.operator`of course, as it was before.

**Pull request merged**